### ISSUE TITLE
TODO: Full test suite according to spec

### ISSUE THREAD

**Issue by Joannis** [OpenKitten/MongoKitten#120]
https://github.com/mongodb/specifications

**Comment by Joannis** [comment]
BSON has this implemented

**Comment by shanehsu** [comment]
By just scanning through the [folder names](https://github.com/mongodb/specifications/tree/master/source), around two belongs to BSON, the rest is MongoKitten territory?

**Comment by Joannis** [comment]
Yes. But the MongoKitten tests are quite tedious to implement. I want to write a tool that takes these JSON tests and generates Swift tests for them. Otherwise it'd take me weeks.

MongoKitten tests already cover a huge chunk of the what the specification covers, so stability wise we're quite decent.

I did do BSON manually, and first. Since BSON is *the* most important part of MongoKitten. Everything hinges on that.
